File: ThirtyDaysOfCodeInJava/scope_14.java
/* Scope Day 14 - https://www.hackerrank.com/challenges/30-scope?h_r=next-challenge&h_v=zen
Complete the Difference class by writing the following:

A class constructor that takes an array of integers as a parameter and saves
it to the  instance variable. A computeDifference method that finds the
maximum absolute difference between any  numbers in and stores it in the
maximumDifference instance variable. */

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;


class Difference {
  	private int[] elements;
  	public int maximumDifference;

    public Difference(int[] elements){
        this.elements = elements;
    }

    public void computeDifference(){
        Arrays.sort(elements);
        maximumDifference = elements[elements.length-1] - elements[0];
    }
}

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        sc.close();

        Difference difference = new Difference(a);

        difference.computeDifference();

        System.out.print(difference.maximumDifference);
    }
}

File: ThirtyDaysOfCodeInPython/scope_14.py
# Scope Day 14 - https://www.hackerrank.com/challenges/30-scope?h_r=next-challenge&h_v=zen
# Complete the Difference class by writing the following:

# A class constructor that takes an array of integers as a parameter and saves
# it to the elements instance variable. A computeDifference method that finds the
# maximum absolute difference between any  numbers in and stores it in the
# maximumDifference instance variable. 


import sys 

class Difference:
	maximumDifference = 0
	elements = []
	def __init__(self,elements):
		self.elements = elements

	def computeDifference(self):
		self.elements = sorted(self.elements)
		self.maximumDifference = self.elements[len(self.elements)-1] - self.elements[0]


a = [-9,3,-5,-1,-2]	
newClass = Difference(a)
newClass.computeDifference()
print(newClass.maximumDifference)